core/ast_node.js

```diff
@@ -445,28 +445,46 @@ Blockly.ASTNode.prototype.findTopASTNodeForBlock_ = function(block) {
   }
 };
 
-
 /**
- * Given a location in a stack of blocks find the next out connection. If the
- * location is nested the next out location should be the connected input.
- * @param {Blockly.Block} block The source block for the current location.
- * @return {Blockly.ASTNode} The ast node holding the next out connection or
- *     block.
+ * Get the ast node pointing to the input that the block is nested under or if
```

Comment:
Now or tomorrow, can you walk me through what happens when the block has an output instead of a previous?

I think it's that `findTopOfSubStack_` returns the block it was passed, and then `topConnection` is the `outputConnection` and it just goes to that input.  But if so, maybe you should only call `findTopOfSubStack_` if you know that the block has a previous connection, and just use `block` directly if you know you have an output.

Reply:
Yup that is what it is doing, and ok sounds good that would definitely make it more clear.